docs/architecture/dapr-for-net-developers/publish-subscribe-buildingblock.md

```diff
@@ -0,0 +1,398 @@
+
+### Using Publish/Subscribe components
+
+Within the eShopOnDapr repository, a `deployment` folder contains files for deploying the application using different deployment modes: `Docker Compose` and `Kubernetes`. A `dapr` folder exists within each of these folders that holds a `components` folder. This folder holds a file `eshop-pubsub.yaml` containing the configuration of the Dapr Publish/Subscribe component that the application will use for Publish/Subscribe behavior. As you saw in the earlier code snippets, the name of the Publish/Subscribe component used is `pubsub`. Here's the content of the `eshop-pubsub.yaml` file in the `deployment/compose/dapr/components` folder:
+
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: pubsub
+  namespace: default
+spec:
+  type: pubsub.nats
+  metadata:
+  - name: natsURL
+    value: nats://demo.nats.io:4222
+```
+
+Note how the [NATS message broker](https://nats.io/) is configured for this example. To change message brokers, you need only to configure a different message broker, such as RabbitMQ or Azure Service Bus and update the yaml file. With Dapr, there are no changes to your mainline service code when switching message brokers.
+
+Finally, you might ask, "Why would I need multiple message brokers in an application?". Many times a system will handle workloads with different characteristics. One event may occur 10 times a day, but another event occurs 5000 times per second. You may benefit by partitioning messaging traffic to different message brokers. With Dapr, you can add multiple Publish/Subscribe component configurations, each with a different name.
+
+## Summary
+
+The Publish/Subscribe pattern helps you decouple services in a distributed application. The Dapr Publish/Subscribe building block simplifies implementing this behavior in your application.
+
+Through the Dapr Publish-Subscribe building block, you can publish messages to a specific *topic*. As well, the building block will query your service to determine which topic(s) to subscribe to.
+
+You can use the Publish/Subscribe building block natively over HTTP or by using one of the language-specific SDKs, such as the .NET SDK for Dapr. The .NET SDK tightly integrates with the ASP.NET core platform.
+
+With Dapr, you can plug a supported message broker product into your application. You can then swap message brokers without requiring code changes to your application.
+
+### References
+
+>[!div class="step-by-step"]
+>[Previous](serviceinvocation.md)
+>[Next](bindings-buildingblock.md)
```

Comment:
```suggestion
> [!div class="step-by-step"]
> [Previous](serviceinvocation.md)
> [Next](bindings-buildingblock.md)
```

This is the navigation of the chapters, not References. Also, these files do not yet exist and are causing build warnings. We might want to comment this out for now.